### PR TITLE
Commit of edited files

### PR DESCRIPTION
**Pull request by KFANullSense**
Had to copy my changes, delete the rep and start again because I messed up the merge. These are the correct files.

### CODE REVIEW

File: CMP1903_A1_2324/Die.cs [hunk 1]
```diff
@@ -8,16 +8,21 @@ namespace CMP1903_A1_2324
...
+        public int Roll()
+        {
+            int rolledValue = RNG.Next(1, 7); //Using the static Random instance, generate a random integer between 1 and 6.
 
-        //Method
-
+            //Set the current value of the dice to the random value, and return it.
+            CurrentValue = rolledValue;
+            return rolledValue;
+        }
```
Comment (Nedkins) [inline CMP1903_A1_2324/Die.cs:22]: Empty line could be removed, however I understand why it was left there to separate the end of the Roll method.  Seeing how this was not done in the other classes however, I would recommend removing this.

File: CMP1903_A1_2324/Testing.cs [hunk 2]
```diff
@@ -3,18 +3,30 @@
...
+            }
+
+            Die localDieInstance = new Die(); //Create a local Die class.
+
+            int rollToTest = localDieInstance.Roll(); //Roll the die.
+
+            Debug.Assert(rollToTest >= 1 && rollToTest <= 6); //Ensure that the outputted value of the die is within expected bounds.
+        }
     }
 }
```
Comment (Nedkins) [inline CMP1903_A1_2324/Testing.cs:31]: For the testing section, the code looks perfect and does the job, however adding a simple console message to let the user know if the tests have been passed or failed (and which test failed), could be beneficial to the user.

Comment (Nedkins) [inline CMP1903_A1_2324/Testing.cs:31]: As an overall, there is not a lot of things that need changing. It seems you have completed every section basically perfectly and your code also seems to follow the general naming guidelines from the Google C# Document. One tiny change that could be done, just to 'clean up', is removing unused imports at the top of each class script. Your code is very easy to read with clear comments that state what the code is doing. You also seem to have any errors handled as when asking the user for an input, you use a loop to make sure the user input is valid.

**Review by Nedkins — COMMENTED** [review]
Make more mistakes next time Sam, this is very difficult!

File: CMP1903_A1_2324/Die.cs [hunk 3]
```diff
@@ -8,16 +8,21 @@ namespace CMP1903_A1_2324
 {
     internal class Die
     {
-        /*
-         * The Die class should contain one property to hold the current die value,
-         * and one method that rolls the die, returns and integer and takes no parameters.
-         */
+        //By making the Random instance static, it ensures that all rolls of the dice will continue along the same seed.
+        public static Random RNG = new Random();
```
Comment (Fnly23) [inline CMP1903_A1_2324/Die.cs:9]: the empty lines could be deleted however not completely necessary

File: CMP1903_A1_2324/Game.cs
Comment (Fnly23) [inline CMP1903_A1_2324/Game.cs:1]: Looks good to me !

File: CMP1903_A1_2324/Program.cs
Comment (Fnly23) [inline CMP1903_A1_2324/Program.cs:1]: Looks good to me !

File: CMP1903_A1_2324/Testing.cs
Comment (Fnly23) [inline CMP1903_A1_2324/Testing.cs:1]: There is no way for the user to check if the testing has worked or not, and if they have failed, which ones failed.

**Review by Fnly23 — COMMENTED** [review]
In total this looks very good. Does the job extremely well and follows the c# google guidlines extremely well.

**Pull request merged**